lib/plan_executor/applicator.rb

```diff
@@ -0,0 +1,30 @@
+# frozen_string_literal: true
+
+require 'bolt/error'
+
+module PlanExecutor
+  class Applicator
+    def initialize(inventory, executor, config)
+      @inventory = inventory
+      @executor = executor
+      @config = config
+    end
+
+    def raise_not_implemented(feature)
+      raise Bolt::Error.new("#{feature} not implemented for plan executor service.",
+                            'bolt.plan-executor/not-implemented')
+    end
+
+    def apply(_args, _apply_body, _scope)
+      raise_not_implemented("apply")
+    end
+
+    def build_plugin_tarball
+      raise_not_implemented("build_plugin_tarball")
+    end
+
+    def custom_facts_task
+      raise_not_implemented('custom_facts_task')
+    end
```

Comment:
Need query_resources_task as well now. Although I'm not sure why you specifically want to avoid those, it seems like we could inherit from `Bolt::Applicator` and only override `apply`.

Reply:
We don't want to upload the plugintarballs over orchestrator.

Reply:
Still makes sense to error on `query_resources_task`, although `build_plugin_tarball` will error first.